larray/io/session.py

```diff
@@ -67,6 +68,18 @@ def close(self):
         """
         raise NotImplementedError()
 
+    def _get_original_file_name(self):
+        original_file_name = None
+        if self.overwrite_file and os.path.isfile(self.fname):
+            original_file_name = self.fname
+            self.fname = '{}~{}'.format(*os.path.splitext(self.fname))
+        return original_file_name
+
+    def _update_original_file(self, original_file_name):
```

Comment:
My original comment about passing state both via arguments and via attributes still stand. I know passing the state via attributes is debatable/is a matter of style in this case, but mixing both styles without a good reason (at least that I can see) feels odd.